internal/fleet/lifecycle.go

```
@@ -0,0 +1,259 @@
+		if err := metadatafile.RequireRealEntry(target, targetInfo, false, "Deployment Profile", "AGX-FLEET-PROFILE-WRITE"); err != nil {
+			return err
+		}
+	} else if !os.IsNotExist(targetErr) {
+		return fmt.Errorf("AGX-FLEET-PROFILE-WRITE: cannot inspect Deployment Profile: %w", targetErr)
+	}
+	temporary, err := os.CreateTemp(directory, ".fleet-profile-*.tmp")
+	if err != nil {
+		return fmt.Errorf("AGX-FLEET-PROFILE-WRITE: %w", err)
+	}
+	temporaryName := temporary.Name()
+	defer os.Remove(temporaryName)
```

golangci-lint (2.12.2)

[error] Error return value of `os.Remove` is not checked

(errcheck)